x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -187,20 +187,32 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 func (m *MetricSet) reportEvent(ms mapstr.M, reporter mb.ReporterV2, qry string) {
 	if m.Config.RawData.Enabled {
 
```

Comment:
Another way to do this would be:
```
moduleFields := mapstr.M{
	"metrics": ms, // Individual metric
	"driver":  m.Config.Driver,
}
if qry != "" {
	moduleFields["query"] = qry
}
reporter.Event(mb.Event{
	ModuleFields: moduleFields
})
```

Reply:
Yes. I hope we can stay with the current structure.

Reply:
I mean, you can do it this way in case you feel it complicates less the code :slightly_smiling_face: https://github.com/elastic/beats/pull/32394#discussion_r926445134